### PR TITLE
Add clearable input functionality and use for configuration panels

### PR DESCRIPTION
**Pull request by toddburnside**
The clear icon only appears when there is something to clear and matches the clear icon for the select controls.
https://user-images.githubusercontent.com/6035943/179288858-c1d6f8d5-406c-446c-92b3-21c77770421f.mov

### CODE REVIEW

File: common/src/main/scala/explore/utils/package.scala [hunk 1]
```diff
@@ -105,4 +107,23 @@ package object utils {
...
+   * Depending on the layout containing the `FormInputEV`, you may also need to add the
+   * `ExploreStyles.ClearableInputPaddingReset` class. When an icon is added to a FormInput, it
+   * changes the padding for the enclosed `input` to make room for the icon - lots of room - which
+   * can change your layout. The above class resets the padding to the standard for FormInputs
+   * without icons. The downside, of course, is that if your content can extend all the wqy to the
+   * right side of the input, the `X` icon will sit on top of it. Usually this is not a problem.
+   * Note that this will override that for `InputWithUnits` this will override the default `clazz`
+   * of `ExploreStyles.Grow(1), so you may need to add that, too.
+   */
+  def clearInputIcon[A](view: View[Option[A]]): js.UndefOr[VdomNode] =
```
Comment (cquiroz) [inline common/src/main/scala/explore/utils/package.scala:125]: Maybe `clearable`? `clear` sounds to me like transparent?

Comment (cquiroz) [reply common/src/main/scala/explore/utils/package.scala:125]: Never mind, it is the icon

Comment (toddburnside) [reply common/src/main/scala/explore/utils/package.scala:125]: Sure. I was thinking of the action here, but I use clearable in other places.

**Review by cquiroz — APPROVED** [review]
Looks great

**Comment by rpiaggio** [comment]
I like a lot the resulting UI but the syntax to define the clearable controls seems a bit convoluted.

I wonder if we can add a `clearable` extension method to `InputWithUnits` when `A =:= Option[_]` ?

File: explore/src/main/scala/explore/config/AdvancedConfigurationPanel.scala [hunk 2]
```diff
@@ -209,36 +210,42 @@ sealed abstract class AdvancedConfigurationPanelBuilder[
               )
               .flatten
 
-          def dithersControl(onChange: Callback): VdomElement =
+          def dithersControl(onChange: Callback): VdomElement = {
+            val view = explicitWavelengthDithers(props.scienceModeAdvanced).withOnMod(_ => onChange)
```
[the remark this answers is not in the document]
Comment (toddburnside) [reply explore/src/main/scala/explore/config/AdvancedConfigurationPanel.scala:214]: True. I'll remove.

**Pull request merged**